// @alipay/mini-api-types/lib/public/apis/media/ar/animation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @description propertyId 对应的动画属性类型。
 */
var EPropertyId;
(function (EPropertyId) {
    /**
     * @description 整体缩放，每帧一个scale值，data = scale
     */
    EPropertyId[EPropertyId["SCALE_UNIT"] = 0] = "SCALE_UNIT";
    /**
     * @description 各分量缩放，每帧三个值，data = sx, sy, sz
     */
    EPropertyId[EPropertyId["SCALE"] = 1] = "SCALE";
    /**
     * @description 缩放x分量，x方向上缩放，data = sx
     */
    EPropertyId[EPropertyId["SCALE_X"] = 2] = "SCALE_X";
    /**
     * @description 缩放y分量，y方向上缩放，data = sy
     */
    EPropertyId[EPropertyId["SCALE_Y"] = 3] = "SCALE_Y";
    /**
     * @description 缩放z分量，z方向上缩放，data = sz
     */
    EPropertyId[EPropertyId["SCALE_Z"] = 4] = "SCALE_Z";
    /**
     * @description 各方向旋转，每帧一个旋转四元数，data = qx, qy, qz, qw
     */
    EPropertyId[EPropertyId["ROTATE"] = 8] = "ROTATE";
    /**
     * @description 各方向位移，每帧一个位移向量，data = tx, ty, tz
     */
    EPropertyId[EPropertyId["TRANSLATE"] = 9] = "TRANSLATE";
    /**
     * @description x方向上位移，data = tx
     */
    EPropertyId[EPropertyId["TRANSLATE_X"] = 10] = "TRANSLATE_X";
    /**
     * @description y方向上位移，data = ty
     */
    EPropertyId[EPropertyId["TRANSLATE_Y"] = 11] = "TRANSLATE_Y";
    /**
     * @description z方向位移，data = tz
     */
    EPropertyId[EPropertyId["TRANSLATE_Z"] = 12] = "TRANSLATE_Z";
    /**
     * @description 旋转、位移，data = qx, qy, qz, qw, tx, ty, tz
     */
    EPropertyId[EPropertyId["ROTATE_TRANSLATE"] = 16] = "ROTATE_TRANSLATE";
    /**
     * @description 缩放、旋转加上位移，data = sx, sy, sz, qx, qy, qz, qw, tx, ty, tz
     */
    EPropertyId[EPropertyId["SCALE_ROTATE_TRANSLATE"] = 17] = "SCALE_ROTATE_TRANSLATE";
    /**
     * @description 缩放、位移，data = sx, sy, sz, tx, ty, tz
     */
    EPropertyId[EPropertyId["SCALE_TRANSLATE"] = 18] = "SCALE_TRANSLATE";
    /**
     * @description 缩放、旋转，data = sx, sy, sz, qx, qy, qz, qw
     */
    EPropertyId[EPropertyId["SCALE_ROTATE"] = 19] = "SCALE_ROTATE";
})(EPropertyId || (EPropertyId = {}));
/**
 * @description 动画插值类型。
 */
var EAnimationType;
(function (EAnimationType) {
    /**
     * @description BSPLINE，B样条曲线动画。
    */
    EAnimationType[EAnimationType["BSPLINE"] = 1] = "BSPLINE";
    /**
     * @description FLAT，平滑动画。
    */
    EAnimationType[EAnimationType["FLAT"] = 2] = "FLAT";
    /**
     * @description LINEAR，线性动画。
    */
    EAnimationType[EAnimationType["LINEAR"] = 4] = "LINEAR";
    /**
     * @description SMOOTH，光滑曲线动画。
    */
    EAnimationType[EAnimationType["SMOOTH"] = 5] = "SMOOTH";
    /**
     * @description STEP，阶梯曲线动画。
    */
    EAnimationType[EAnimationType["STEP"] = 6] = "STEP";
    /**
     * @description QUADRATIC_IN。
    */
    EAnimationType[EAnimationType["QUADRATIC_IN"] = 7] = "QUADRATIC_IN";
    /**
     * @description QUADRATIC_OUT。
    */
    EAnimationType[EAnimationType["QUADRATIC_OUT"] = 8] = "QUADRATIC_OUT";
    /**
     * @description QUADRATIC_IN_OUT。
    */
    EAnimationType[EAnimationType["QUADRATIC_IN_OUT"] = 9] = "QUADRATIC_IN_OUT";
    /**
     * @description QUADRATIC_OUT_IN。
    */
    EAnimationType[EAnimationType["QUADRATIC_OUT_IN"] = 10] = "QUADRATIC_OUT_IN";
    /**
     * @description CUBIC_IN。
    */
    EAnimationType[EAnimationType["CUBIC_IN"] = 11] = "CUBIC_IN";
    /**
     * @description CUBIC_OUT。
    */
    EAnimationType[EAnimationType["CUBIC_OUT"] = 12] = "CUBIC_OUT";
    /**
     * @description CUBIC_IN_OUT。
    */
    EAnimationType[EAnimationType["CUBIC_IN_OUT"] = 13] = "CUBIC_IN_OUT";
    /**
     * @description CUBIC_OUT_IN。
    */
    EAnimationType[EAnimationType["CUBIC_OUT_IN"] = 14] = "CUBIC_OUT_IN";
    /**
     * @description QUARTIC_IN。
    */
    EAnimationType[EAnimationType["QUARTIC_IN"] = 15] = "QUARTIC_IN";
    /**
     * @description QUARTIC_OUT。
    */
    EAnimationType[EAnimationType["QUARTIC_OUT"] = 16] = "QUARTIC_OUT";
    /**
     * @description QUARTIC_IN_OUT。
    */
    EAnimationType[EAnimationType["QUARTIC_IN_OUT"] = 17] = "QUARTIC_IN_OUT";
    /**
     * @description QUARTIC_OUT_IN。
    */
    EAnimationType[EAnimationType["QUARTIC_OUT_IN"] = 18] = "QUARTIC_OUT_IN";
    /**
     * @description QUINTIC_IN。
    */
    EAnimationType[EAnimationType["QUINTIC_IN"] = 19] = "QUINTIC_IN";
    /**
     * @description QUINTIC_OUT。
    */
    EAnimationType[EAnimationType["QUINTIC_OUT"] = 20] = "QUINTIC_OUT";
    /**
     * @description QUINTIC_IN_OUT。
    */
    EAnimationType[EAnimationType["QUINTIC_IN_OUT"] = 21] = "QUINTIC_IN_OUT";
    /**
     * @description QUINTIC_OUT_IN。
    */
    EAnimationType[EAnimationType["QUINTIC_OUT_IN"] = 22] = "QUINTIC_OUT_IN";
    /**
     * @description SINE_IN。
    */
    EAnimationType[EAnimationType["SINE_IN"] = 23] = "SINE_IN";
    /**
     * @description SINE_OUT。
    */
    EAnimationType[EAnimationType["SINE_OUT"] = 24] = "SINE_OUT";
    /**
     * @description SINE_IN_OUT。
    */
    EAnimationType[EAnimationType["SINE_IN_OUT"] = 25] = "SINE_IN_OUT";
    /**
     * @description SINE_OUT_IN。
    */
    EAnimationType[EAnimationType["SINE_OUT_IN"] = 26] = "SINE_OUT_IN";
    /**
     * @description EXPONENTIAL_IN。
    */
    EAnimationType[EAnimationType["EXPONENTIAL_IN"] = 27] = "EXPONENTIAL_IN";
    /**
     * @description EXPONENTIAL_OUT。
    */
    EAnimationType[EAnimationType["EXPONENTIAL_OUT"] = 28] = "EXPONENTIAL_OUT";
    /**
     * @description EXPONENTIAL_IN_OUT。
    */
    EAnimationType[EAnimationType["EXPONENTIAL_IN_OUT"] = 29] = "EXPONENTIAL_IN_OUT";
    /**
     * @description EXPONENTIAL_OUT_IN。
    */
    EAnimationType[EAnimationType["EXPONENTIAL_OUT_IN"] = 30] = "EXPONENTIAL_OUT_IN";
    /**
     * @description CIRCULAR_IN。
    */
    EAnimationType[EAnimationType["CIRCULAR_IN"] = 31] = "CIRCULAR_IN";
    /**
     * @description CIRCULAR_OUT。
    */
    EAnimationType[EAnimationType["CIRCULAR_OUT"] = 32] = "CIRCULAR_OUT";
    /**
     * @description CIRCULAR_IN_OUT。
    */
    EAnimationType[EAnimationType["CIRCULAR_IN_OUT"] = 33] = "CIRCULAR_IN_OUT";
    /**
     * @description CIRCULAR_OUT_IN。
    */
    EAnimationType[EAnimationType["CIRCULAR_OUT_IN"] = 34] = "CIRCULAR_OUT_IN";
    /**
     * @description ELASTIC_IN。
    */
    EAnimationType[EAnimationType["ELASTIC_IN"] = 35] = "ELASTIC_IN";
    /**
     * @description ELASTIC_OUT。
    */
    EAnimationType[EAnimationType["ELASTIC_OUT"] = 36] = "ELASTIC_OUT";
    /**
     * @description ELASTIC_IN_OUT。
    */
    EAnimationType[EAnimationType["ELASTIC_IN_OUT"] = 37] = "ELASTIC_IN_OUT";
    /**
     * @description ELASTIC_OUT_IN。
    */
    EAnimationType[EAnimationType["ELASTIC_OUT_IN"] = 38] = "ELASTIC_OUT_IN";
    /**
     * @description OVERSHOOT_IN。
    */
    EAnimationType[EAnimationType["OVERSHOOT_IN"] = 39] = "OVERSHOOT_IN";
    /**
     * @description OVERSHOOT_OUT。
    */
    EAnimationType[EAnimationType["OVERSHOOT_OUT"] = 40] = "OVERSHOOT_OUT";
    /**
     * @description OVERSHOOT_IN_OUT。
    */
    EAnimationType[EAnimationType["OVERSHOOT_IN_OUT"] = 41] = "OVERSHOOT_IN_OUT";
    /**
     * @description OVERSHOOT_OUT_IN。
    */
    EAnimationType[EAnimationType["OVERSHOOT_OUT_IN"] = 42] = "OVERSHOOT_OUT_IN";
    /**
     * @description BOUNCE_IN。
    */
    EAnimationType[EAnimationType["BOUNCE_IN"] = 43] = "BOUNCE_IN";
    /**
     * @description BOUNCE_OUT。
    */
    EAnimationType[EAnimationType["BOUNCE_OUT"] = 44] = "BOUNCE_OUT";
    /**
     * @description BOUNCE_IN_OUT。
    */
    EAnimationType[EAnimationType["BOUNCE_IN_OUT"] = 45] = "BOUNCE_IN_OUT";
    /**
     * @description BOUNCE_OUT_IN。
    */
    EAnimationType[EAnimationType["BOUNCE_OUT_IN"] = 46] = "BOUNCE_OUT_IN";
})(EAnimationType || (EAnimationType = {}));
