// @alipay/mini-api-types/lib/public/apis/media/ar/gesture-detect.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EGestureDetectStatus = exports.EGestureDetectLabel = void 0;
/**
 * @description 标签命名规则：f表示手掌对着摄像头，b表示手背对着摄像头；从小拇指到大拇指编号依次为1～5，比如Gesture-f34表示手掌对着摄像头，露出中指 (编号3) 和食指 (编号4)。
 */
var EGestureDetectLabel;
(function (EGestureDetectLabel) {
    /**
     * @description ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198053214-ba782d35-6919-4ff6-b557-4473b72160f6.png)
     */
    EGestureDetectLabel["Gesture-b45"] = "Gesture-b45";
    /**
     * @description ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198074533-5340b2b9-7026-4d7e-920b-c67457fe48c9.png)
     */
    EGestureDetectLabel["Gesture-f45"] = "Gesture-f45";
    /**
     * @description ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198127821-0647c8ee-97d5-4da4-b814-b43e342117cb.png)
     */
    EGestureDetectLabel["Gesture-fAll"] = "Gesture-fAll";
    /**
     * @description ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198230144-32f4dc74-5358-4a71-9666-a0babe7a28a5.png)
     */
    EGestureDetectLabel["Gesture-bAll"] = "Gesture-bAll";
    /**
     * @description ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198271555-33998ae7-3ada-429f-a163-a1c18c765a55.png)
     */
    EGestureDetectLabel["Gesture-f34"] = "Gesture-f34";
    /**
     * @description ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198291400-063a0013-a14d-4f16-bf92-762acc813916.png)
     */
    EGestureDetectLabel["Gesture-b34"] = "Gesture-b34";
    /**
     * @description
     * - ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198390489-5736de1d-e35d-4066-a455-278cf3793225.png)
     * - ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198376341-d213c641-3324-4ae1-a65b-406647124e91.png)
     */
    EGestureDetectLabel["Gesture-f0"] = "Gesture-f0";
    /**
     * @description ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198403067-3c826470-7ba7-45d2-b657-59b722c906bc.png)
     */
    EGestureDetectLabel["Gesture-d0"] = "Gesture-d0";
    /**
     * @description ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198420983-3a9dad04-f458-4a8c-8587-2d356fa786fc.png)
     */
    EGestureDetectLabel["Gesture-b15"] = "Gesture-b15";
    /**
     * @description ![image](https://intranetproxy.alipay.com/skylark/lark/0/2019/png/24428/1554198432918-d1a4edb4-f1dc-4934-8b6e-ce96bda2cf81.png)
     */
    EGestureDetectLabel["Gesture-f15"] = "Gesture-f15";
})(EGestureDetectLabel = exports.EGestureDetectLabel || (exports.EGestureDetectLabel = {}));
/**
 * @description 手势识别引擎状态。
 */
var EGestureDetectStatus;
(function (EGestureDetectStatus) {
    /**
     * @description 设备不支持。
     */
    EGestureDetectStatus["NotSupport"] = "NotSupport";
    /**
     * @description 正在初始化。
     */
    EGestureDetectStatus["Initializing"] = "Initializing";
    /**
     * @description 准备就绪。
     */
    EGestureDetectStatus["Ready"] = "Ready";
    /**
     * @description 未知。
     */
    EGestureDetectStatus["Unknow"] = "Unknow";
})(EGestureDetectStatus = exports.EGestureDetectStatus || (exports.EGestureDetectStatus = {}));
