// @alipay/mini-api-types/lib/public/apis/media/camera/ai-camera.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @description 图片质量
 */
var EQuality;
(function (EQuality) {
    /**
     * @description 高质量。
     */
    EQuality["high"] = "high";
    /**
     * @description 普通质量。
     */
    EQuality["normal"] = "normal";
    /**
     * @description 低质量。
     */
    EQuality["low"] = "low";
})(EQuality || (EQuality = {}));
var EBizType;
(function (EBizType) {
    /**
     * @description 通用检测
     */
    EBizType["commonDetect"] = "commonDetect";
    /**
     * @description 通用分类
     */
    EBizType["commonClassify"] = "commonClassify";
    /**
     * @description 通用预测
     */
    EBizType["commonPredict"] = "commonPredict";
    /**
     * @description 通用OCR
     */
    EBizType["commonOcr"] = "commonOcr";
    /**
     * @description 驾驶证正面
     */
    EBizType["driverLicenseFront"] = "driverLicenseFront";
    /**
     * @description 驾驶证反面
     */
    EBizType["driverLicenseBack"] = "driverLicenseBack";
    /**
     * @description 行驶证正面
     */
    EBizType["vehicleFront"] = "vehicleFront";
    /**
     * @description 行驶证反面
     */
    EBizType["vehicleBack"] = "vehicleBack";
    /**
     * @description 身份证正面
     */
    EBizType["idcardFront"] = "idcardFront";
    /**
     * @description 身份证反面
     */
    EBizType["idcardBack"] = "idcardBack";
    /**
     * @description 银行卡
     */
    EBizType["bankcard"] = "bankcard";
    /**
     * @description 车牌
     */
    EBizType["vehiclePlate"] = "vehiclePlate";
    /**
     * @description 汽车VIN码
     */
    EBizType["vinCode"] = "vinCode";
    /**
     * @description 燃气表示数识别
     */
    EBizType["gasMeters"] = "gasMeters";
    /**
     * @description 护照识别
     */
    EBizType["passport"] = "passport";
})(EBizType || (EBizType = {}));
