// @alipay/mini-api-types/lib/public/apis/media/camera/ocr.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var EOCRType;
(function (EOCRType) {
    /**
     * @description 银行卡识别
     */
    EOCRType["ocr_bank_card"] = "ocr_bank_card";
    /**
     * @description 名片识别
     */
    EOCRType["ocr_business_card"] = "ocr_business_card";
    /**
     * @description 营业执照识别
     */
    EOCRType["ocr_business_license"] = "ocr_business_license";
    /**
     * @description 驾驶证识别，face/back 主页/副页
     */
    EOCRType["ocr_driver_license"] = "ocr_driver_license";
    /**
     * @description 身份证识别，face/back 正/反
     */
    EOCRType["ocr_idcard"] = "ocr_idcard";
    /**
     * @description 护照识别
     */
    EOCRType["ocr_passport"] = "ocr_passport";
    /**
     * @description 火车票识别
     */
    EOCRType["ocr_train_ticket"] = "ocr_train_ticket";
    /**
     * @description 行驶证识别
     */
    EOCRType["ocr_vehicle"] = "ocr_vehicle";
    /**
     * @description 车牌识别
     */
    EOCRType["ocr_vehicle_plate"] = "ocr_vehicle_plate";
    /**
     * @description 通用文字识别
     */
    EOCRType["ocr_general"] = "ocr_general";
    /**
     * @description vin识别
     */
    EOCRType["ocr_vin"] = "ocr_vin";
})(EOCRType || (EOCRType = {}));
var EXnnOcrType;
(function (EXnnOcrType) {
    /**
     * @description 身份证正面
     */
    EXnnOcrType["identityCardAvatar"] = "identityCardAvatar";
    /**
     * @description 身份证背面
     */
    EXnnOcrType["identityCardEmblem"] = "identityCardEmblem";
    /**
     * @description 银行卡
     */
    EXnnOcrType["bankcard"] = "bankcard";
    /**
     * @description 车牌
     */
    EXnnOcrType["plate"] = "plate";
    /**
     * @description vin码
     */
    EXnnOcrType["vinCode"] = "vinCode";
})(EXnnOcrType || (EXnnOcrType = {}));
var EXnnOcrErrorCode;
(function (EXnnOcrErrorCode) {
    /**
     * @description 输入参数错误。
     */
    EXnnOcrErrorCode[EXnnOcrErrorCode["_1"] = 1] = "_1";
    /**
     * @description 机型不支持或者被降级。
     */
    EXnnOcrErrorCode[EXnnOcrErrorCode["_2"] = 2] = "_2";
    /**
     * @description 未处理，可能引擎处于busy状态。
     */
    EXnnOcrErrorCode[EXnnOcrErrorCode["_3"] = 3] = "_3";
    /**
     * @description 算法引擎初始化/处理等内部错误。
     */
    EXnnOcrErrorCode[EXnnOcrErrorCode["_4"] = 4] = "_4";
    /**
     * @description 内部配置信息拉取/解析失败。
     */
    EXnnOcrErrorCode[EXnnOcrErrorCode["_5"] = 5] = "_5";
    /**
     * @description 模型文件未下载或相关错误。
     */
    EXnnOcrErrorCode[EXnnOcrErrorCode["_6"] = 6] = "_6";
    /**
     * @description 内存太小。
     */
    EXnnOcrErrorCode[EXnnOcrErrorCode["_7"] = 7] = "_7";
    /**
     * @description CPU太高。
     */
    EXnnOcrErrorCode[EXnnOcrErrorCode["_8"] = 8] = "_8";
})(EXnnOcrErrorCode || (EXnnOcrErrorCode = {}));
