// @alipay/mini-api-types/lib/public/apis/media/speech/recognize.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ESpeechRecognizeErrorCode;
(function (ESpeechRecognizeErrorCode) {
    /**
     * @description 【通用错误码】成功
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ERROR_NONE"] = 0] = "ERROR_NONE";
    /**
     * @description 【通用错误码】输入参数错误
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ERROR_INVALID_PARAMS"] = 1] = "ERROR_INVALID_PARAMS";
    /**
     * @description 【语音识别错误】签名错误/没有签名
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["SERVICE_ERROR_NO_SIGNATURE"] = 10] = "SERVICE_ERROR_NO_SIGNATURE";
    /**
     * @description 【语音识别错误】无效参数
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["SERVICE_ERROR_PARAM_INVALID"] = 12] = "SERVICE_ERROR_PARAM_INVALID";
    /**
     * @description 【语音识别错误】未知错误
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["START_ERROR_UNKNOWN"] = 21] = "START_ERROR_UNKNOWN";
    /**
     * @description 【语音识别错误】状态无效，正在识别，请先停止
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["START_ERROR_STATE_INVALID"] = 22] = "START_ERROR_STATE_INVALID";
    /**
     * @description 【语音识别错误】识别错误
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["RECOGNIZE_ERROR_SERVER_RECOGNIZE_FAILED"] = 30] = "RECOGNIZE_ERROR_SERVER_RECOGNIZE_FAILED";
    /**
     * @description 【语音识别错误】识别后台错误
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["RECOGNIZE_ERROR_SERVER_INVALID"] = 31] = "RECOGNIZE_ERROR_SERVER_INVALID";
    /**
     * @description 【语音识别错误】网络错误
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["RECOGNIZE_ERROR_NET_ERROR"] = 32] = "RECOGNIZE_ERROR_NET_ERROR";
    /**
     * @description 【语音识别错误】音频opus编码错误
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["RECOGNIZE_ERROR_RECORD_ERROR"] = 33] = "RECOGNIZE_ERROR_RECORD_ERROR";
    /**
     * @description 【语音识别错误】其他错误
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["RECOGNIZE_ERROR_OTHER_ERROR"] = 34] = "RECOGNIZE_ERROR_OTHER_ERROR";
    /**
     * @description 【语音识别错误】取消识别
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["RECOGNIZE_ERROR_CANCEL"] = 38] = "RECOGNIZE_ERROR_CANCEL";
    /**
     * @description 【语音识别错误】用户静音
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["RECOGNIZE_ERROR_VAD_FAILED"] = 40] = "RECOGNIZE_ERROR_VAD_FAILED";
    /**
     * @description 【语音识别错误】识别超时
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["RECOGNIZE_ERROR_SESSION_TIMEOUT"] = 41] = "RECOGNIZE_ERROR_SESSION_TIMEOUT";
    /**
     * @description 【音频采集错误】采集失败
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ERROR_AUDIO_CAPTURE"] = 71] = "ERROR_AUDIO_CAPTURE";
    /**
     * @description 【音频采集错误】无音频采集权限
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ERROR_AUDIO_PERMISSION"] = 72] = "ERROR_AUDIO_PERMISSION";
    /**
     * @description 【音频采集错误】初始化失败
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ERROR_AUDIO_INIT"] = 73] = "ERROR_AUDIO_INIT";
    /**
     * @description 【音频采集错误】音频设备占用
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ERROR_AUDIO_IN_USING"] = 74] = "ERROR_AUDIO_IN_USING";
    /**
     * @description 【音频编码错误】编码失败
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ENCODE_ERROR_ENCODING"] = 91] = "ENCODE_ERROR_ENCODING";
    /**
     * @description 【音频编码错误】关闭失败
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ENCODE_ERROR_ENCODE_CLOSE"] = 92] = "ENCODE_ERROR_ENCODE_CLOSE";
    /**
     * @description 【音频编码错误】写文件失败
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ENCODE_ERROR_WRITE_FILE"] = 93] = "ENCODE_ERROR_WRITE_FILE";
    /**
     * @description 【音频编码错误】长度非法
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ENCODE_ERROR_LENGTH_LESS_ZERO"] = 94] = "ENCODE_ERROR_LENGTH_LESS_ZERO";
    /**
     * @description 【音频编码错误】参数设置非法
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ENCODE_ERROR_OPEN_PARAM"] = 95] = "ENCODE_ERROR_OPEN_PARAM";
    /**
     * @description 【音频编码错误】未知错误
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ENCODE_ERROR_OPEN_UNKNOW"] = 96] = "ENCODE_ERROR_OPEN_UNKNOW";
    /**
     * @description 【音频编码错误】结束失败
     */
    ESpeechRecognizeErrorCode[ESpeechRecognizeErrorCode["ENCODE_ERROR_END_ENCODER"] = 97] = "ENCODE_ERROR_END_ENCODER";
})(ESpeechRecognizeErrorCode || (ESpeechRecognizeErrorCode = {}));
