// @alipay/mini-api-types/lib/public/apis/media/canvas.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var EFileType;
(function (EFileType) {
    /**
     * @description jpg 图片。
     */
    EFileType["Jpg"] = "jpg";
    /**
     * @description png 图片。
     */
    EFileType["Png"] = "png";
})(EFileType || (EFileType = {}));
var ELineCap;
(function (ELineCap) {
    ELineCap["Round"] = "round";
    ELineCap["Butt"] = "butt";
    ELineCap["Square"] = "square";
})(ELineCap || (ELineCap = {}));
var ELineJoin;
(function (ELineJoin) {
    ELineJoin["Round"] = "round";
    ELineJoin["Bevel"] = "bevel";
    ELineJoin["Miter"] = "miter";
})(ELineJoin || (ELineJoin = {}));
var ETextAlign;
(function (ETextAlign) {
    ETextAlign["Left"] = "left";
    ETextAlign["Right"] = "right";
    ETextAlign["Center"] = "center";
    ETextAlign["Start"] = "start";
    ETextAlign["End"] = "end";
})(ETextAlign || (ETextAlign = {}));
var ETextBaseLine;
(function (ETextBaseLine) {
    ETextBaseLine["Top"] = "top";
    ETextBaseLine["Hanging"] = "hanging";
    ETextBaseLine["Middle"] = "middle";
    ETextBaseLine["Alphabetic"] = "alphabetic";
    ETextBaseLine["Ideographic"] = "ideographic";
    ETextBaseLine["Bottom"] = "bottom";
})(ETextBaseLine || (ETextBaseLine = {}));
