// @alipay/mini-api-types/lib/public/apis/media/image.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var EImageChoosenSizeType;
(function (EImageChoosenSizeType) {
    /**
     * @description 原图
     */
    EImageChoosenSizeType["original"] = "original";
    /**
     * @description 压缩图
     */
    EImageChoosenSizeType["compressed"] = "compressed";
})(EImageChoosenSizeType || (EImageChoosenSizeType = {}));
var EImageChoosenSourceType;
(function (EImageChoosenSourceType) {
    /**
     * @description 拍照
     */
    EImageChoosenSourceType["camera"] = "camera";
    /**
     * @description 相册选取
     */
    EImageChoosenSourceType["album"] = "album";
})(EImageChoosenSourceType || (EImageChoosenSourceType = {}));
var saveImageToPhotosAlbumErrorCode;
(function (saveImageToPhotosAlbumErrorCode) {
    /**
     * @description 参数无效，没有传 url 参数
     */
    saveImageToPhotosAlbumErrorCode[saveImageToPhotosAlbumErrorCode["_2"] = 2] = "_2";
    /**
     * @description 没有开启相册权限(ios only)
     */
    saveImageToPhotosAlbumErrorCode[saveImageToPhotosAlbumErrorCode["_15"] = 15] = "_15";
    /**
     * @description 手机相册存储空间不足(ios only)
     */
    saveImageToPhotosAlbumErrorCode[saveImageToPhotosAlbumErrorCode["_16"] = 16] = "_16";
    /**
     * @description 保存图片过程中的其他错误
     */
    saveImageToPhotosAlbumErrorCode[saveImageToPhotosAlbumErrorCode["_17"] = 17] = "_17";
})(saveImageToPhotosAlbumErrorCode || (saveImageToPhotosAlbumErrorCode = {}));
var ECorrectLevel;
(function (ECorrectLevel) {
    ECorrectLevel["O"] = "O";
    ECorrectLevel["M"] = "M";
    ECorrectLevel["L"] = "L";
    /**
     * @description 默认值, 最高质量
     */
    ECorrectLevel["H"] = "H";
})(ECorrectLevel || (ECorrectLevel = {}));
var EOrientation;
(function (EOrientation) {
    /**
     * @description 默认。
     */
    EOrientation["up"] = "up";
    /**
     * @description 180 度旋转。
     */
    EOrientation["down"] = "down";
    /**
     * @description 逆时针旋转 90 度。
     */
    EOrientation["left"] = "left";
    /**
     * @description 顺时针旋转 90 度。
     */
    EOrientation["right"] = "right";
    /**
     * @description 同 up，但水平翻转。
     */
    EOrientation["up-mirrored"] = "up-mirrored";
    /**
     * @description 同 down，但水平翻转。
     */
    EOrientation["down-mirrored"] = "down-mirrored";
    /**
     * @description 同 left，但垂直翻转。
     */
    EOrientation["left-mirrored"] = "left-mirrored";
    /**
     * @description 同 right，但垂直翻转。
     */
    EOrientation["right-mirrored"] = "right-mirrored";
})(EOrientation || (EOrientation = {}));
