// @alipay/mini-api-types/lib/public/type/meta/api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.API = void 0;
var API;
(function (API) {
    // 应用
    // 基础能力
    API["arrayBufferToBase64"] = "arrayBufferToBase64";
    API["base64ToArrayBuffer"] = "base64ToArrayBuffer";
    //   版本管理
    API["getRunScene"] = "getRunScene";
    API["getUpdateManager"] = "getUpdateManager";
    API["ap$updateAlipayClient"] = "ap.updateAlipayClient";
    API["SDKVersion"] = "SDKVersion";
    API["isIDE"] = "isIDE";
    //   应用信息
    API["getLaunchOptionsSync"] = "getLaunchOptionsSync";
    API["getEnterOptionsSync"] = "getEnterOptionsSync";
    API["getAppIdSync"] = "getAppIdSync";
    //   生命周期
    API["onAppShow"] = "onAppShow";
    API["offAppShow"] = "offAppShow";
    API["onAppHide"] = "onAppHide";
    API["offAppHide"] = "offAppHide";
    API["exit"] = "exit";
    API["onPageNotFound"] = "onPageNotFound";
    API["offPageNotFound"] = "offPageNotFound";
    //   异常监控
    API["onUnhandledRejection"] = "onUnhandledRejection";
    API["offUnhandledRejection"] = "offUnhandledRejection";
    API["onError"] = "onError";
    API["offError"] = "offError";
    API["onComponentError"] = "onComponentError";
    API["offComponentError"] = "offComponentError";
    //   埋点上报
    API["reportAnalytics"] = "reportAnalytics";
    API["reportCustomEvent"] = "reportCustomEvent";
    API["reportCustomError"] = "reportCustomError";
    API["getChinfoChain"] = "getChinfoChain";
    API["appendChinfoChain"] = "appendChinfoChain";
    API["reportBizReady"] = "reportBizReady";
    // 导航路由
    //   应用内页面跳转
    API["navigateBack"] = "navigateBack";
    API["navigateTo"] = "navigateTo";
    API["reLaunch"] = "reLaunch";
    API["redirectTo"] = "redirectTo";
    API["switchTab"] = "switchTab";
    // 视图界面
    //   节点操作
    API["createAnimation"] = "createAnimation";
    API["createSelectorQuery"] = "createSelectorQuery";
    API["createIntersectionObserver"] = "createIntersectionObserver";
    //   字体
    API["loadFontFace"] = "loadFontFace";
    //   滚动
    API["pageScrollTo"] = "pageScrollTo";
    //   下拉刷新
    API["stopPullDownRefresh"] = "stopPullDownRefresh";
    API["setCanPullDown"] = "setCanPullDown";
    API["startPullDownRefresh"] = "startPullDownRefresh";
    API["setViewTop"] = "setViewTop";
    //   窗口背景
    API["setBackgroundColor"] = "setBackgroundColor";
    API["setBackgroundTextStyle"] = "setBackgroundTextStyle";
    /**
     * @deprecated
     */
    API["setWebViewTop"] = "setWebViewTop";
    /**
     * @deprecated
     */
    API["setBackgroundImage"] = "setBackgroundImage";
    /**
     * @deprecated
     */
    API["removeBackgroundImage"] = "removeBackgroundImage";
    //   TabBar
    API["addTabBarItem"] = "addTabBarItem";
    API["removeTabBarItem"] = "removeTabBarItem";
    API["hideTabBar"] = "hideTabBar";
    API["showTabBar"] = "showTabBar";
    API["setTabBarItem"] = "setTabBarItem";
    API["setTabBarStyle"] = "setTabBarStyle";
    API["hideTabBarRedDot"] = "hideTabBarRedDot";
    API["showTabBarRedDot"] = "showTabBarRedDot";
    API["removeTabBarBadge"] = "removeTabBarBadge";
    API["setTabBarBadge"] = "setTabBarBadge";
    //   导航栏
    //     标题
    API["setTransparentTitle"] = "setTransparentTitle";
    API["setNavigationBar"] = "setNavigationBar";
    API["getTitleColor"] = "getTitleColor";
    API["onTitleCloseClick"] = "onTitleCloseClick";
    API["offTitleCloseClick"] = "offTitleCloseClick";
    API["onTitleMoreClick"] = "onTitleMoreClick";
    API["offTitleMoreClick"] = "offTitleMoreClick";
    API["showNavigationBarLoading"] = "showNavigationBarLoading";
    API["hideNavigationBarLoading"] = "hideNavigationBarLoading";
    API["hideDebugVersion"] = "hideDebugVersion";
    //     按钮
    API["setBackButton"] = "setBackButton";
    API["showBackToHomepage"] = "showBackToHomepage";
    API["hideBackHome"] = "hideBackHome";
    API["setCustomPopMenu"] = "setCustomPopMenu";
    API["hideAddToDesktopMenu"] = "hideAddToDesktopMenu";
    API["hideAllAddToDesktopMenu"] = "hideAllAddToDesktopMenu";
    API["setOptionMenu"] = "setOptionMenu";
    API["getMenuButtonBoundingClientRect"] = "getMenuButtonBoundingClientRect";
    //   小程序设置
    API["openSetting"] = "openSetting";
    API["ap$openSetting"] = "ap.openSetting";
    API["getSetting"] = "getSetting";
    API["ap$getSetting"] = "ap.getSetting";
    //   交互反馈
    API["alert"] = "alert";
    API["confirm"] = "confirm";
    API["prompt"] = "prompt";
    API["showLoading"] = "showLoading";
    API["hideLoading"] = "hideLoading";
    API["showToast"] = "showToast";
    API["hideToast"] = "hideToast";
    //   键盘
    API["hideKeyboard"] = "hideKeyboard";
    API["getSelectedTextRange"] = "getSelectedTextRange";
    //   多选项
    API["showActionSheet"] = "showActionSheet";
    API["multiLevelSelect"] = "multiLevelSelect";
    API["optionsSelect"] = "optionsSelect";
    API["datePicker"] = "datePicker";
    // 组件
    //   画布(2D)
    API["createCanvasContext"] = "createCanvasContext";
    API["canvasToTempFilePath"] = "canvasToTempFilePath";
    API["preloadCanvasImage"] = "preloadCanvasImage";
    // 统一放到多媒体吧。。
    API["createLottieContext"] = "createLottieContext";
    API["createWebViewContext"] = "createWebViewContext";
    API["createRtcRoomContext"] = "createRtcRoomContext";
    // 设备
    // mdns
    API["onLocalServiceResolveFail"] = "onLocalServiceResolveFail";
    API["onLocalServiceLost"] = "onLocalServiceLost";
    API["onLocalServiceFound"] = "onLocalServiceFound";
    API["onLocalServiceDiscoveryStop"] = "onLocalServiceDiscoveryStop";
    API["offLocalServiceResolveFail"] = "offLocalServiceResolveFail";
    API["offLocalServiceLost"] = "offLocalServiceLost";
    API["offLocalServiceFound"] = "offLocalServiceFound";
    API["offLocalServiceDiscoveryStop"] = "offLocalServiceDiscoveryStop";
    API["stopLocalServiceDiscovery"] = "stopLocalServiceDiscovery";
    API["startLocalServiceDiscovery"] = "startLocalServiceDiscovery";
    //   服务端时间
    API["getServerTime"] = "getServerTime";
    //   标识符
    API["getDeviceID"] = "getDeviceID";
    API["getDeviceInfo"] = "getDeviceInfo";
    //   系统信息
    API["env"] = "env";
    API["getSystemInfo"] = "getSystemInfo";
    API["getSystemInfoSync"] = "getSystemInfoSync";
    //   联系人
    API["chooseContact"] = "chooseContact";
    API["choosePhoneContact"] = "choosePhoneContact";
    API["chooseAlipayContact"] = "chooseAlipayContact";
    API["addPhoneContact"] = "addPhoneContact";
    //   运动传感器
    API["onGyroscopeChange"] = "onGyroscopeChange";
    API["offGyroscopeChange"] = "offGyroscopeChange";
    API["onAccelerometerChange"] = "onAccelerometerChange";
    API["offAccelerometerChange"] = "offAccelerometerChange";
    API["onCompassChange"] = "onCompassChange";
    API["offCompassChange"] = "offCompassChange";
    API["onDeviceMotionChange"] = "onDeviceMotionChange";
    API["offDeviceMotionChange"] = "offDeviceMotionChange";
    //   振动
    API["vibrate"] = "vibrate";
    API["watchShake"] = "watchShake";
    API["vibrateLong"] = "vibrateLong";
    API["vibrateShort"] = "vibrateShort";
    //   屏幕
    //     亮度
    API["getScreenBrightness"] = "getScreenBrightness";
    API["setScreenBrightness"] = "setScreenBrightness";
    API["setKeepScreenOn"] = "setKeepScreenOn";
    //     截屏
    API["onUserCaptureScreen"] = "onUserCaptureScreen";
    API["offUserCaptureScreen"] = "offUserCaptureScreen";
    //     朝向
    API["setScreenOrientation"] = "setScreenOrientation";
    API["getScreenOrientation"] = "getScreenOrientation";
    //   剪贴板
    API["setClipboard"] = "setClipboard";
    API["getClipboard"] = "getClipboard";
    //   电话短信
    API["makePhoneCall"] = "makePhoneCall";
    API["sendSMS"] = "sendSMS";
    //   日历
    API["addEventCal"] = "addEventCal";
    API["addEventCalendar"] = "addEventCalendar";
    //   电源
    API["getBatteryInfo"] = "getBatteryInfo";
    API["getBatteryInfoSync"] = "getBatteryInfoSync";
    API["onLowPowerWarning"] = "onLowPowerWarning";
    API["offLowPowerWarning"] = "offLowPowerWarning";
    API["isLowPowerMode"] = "isLowPowerMode";
    //   内存
    API["onMemoryWarning"] = "onMemoryWarning";
    API["offMemoryWarning"] = "offMemoryWarning";
    //   网络状态
    API["getCarrierName"] = "getCarrierName";
    API["getNetworkType"] = "getNetworkType";
    API["onNetworkStatusChange"] = "onNetworkStatusChange";
    API["offNetworkStatusChange"] = "offNetworkStatusChange";
    //   WiFi 连接
    API["startWifi"] = "startWifi";
    API["stopWifi"] = "stopWifi";
    API["connectWifi"] = "connectWifi";
    API["getWifiList"] = "getWifiList";
    API["setWifiList"] = "setWifiList";
    API["getConnectedWifi"] = "getConnectedWifi";
    API["onGetWifiList"] = "onGetWifiList";
    API["offGetWifiList"] = "offGetWifiList";
    API["onWifiConnected"] = "onWifiConnected";
    API["offWifiConnected"] = "offWifiConnected";
    API["getWifiBroadcastInfo"] = "getWifiBroadcastInfo";
    API["registerSSID"] = "registerSSID";
    API["unregisterSSID"] = "unregisterSSID";
    //   NFC
    //      交通卡
    API["showAuthenticDialog"] = "showAuthenticDialog";
    API["seNFCServiceIsv"] = "seNFCServiceIsv";
    //     模拟卡
    API["getHCEState"] = "getHCEState";
    API["startHCE"] = "startHCE";
    API["stopHCE"] = "stopHCE";
    API["onHCEMessage"] = "onHCEMessage";
    API["offHCEMessage"] = "offHCEMessage";
    API["sendHCEMessage"] = "sendHCEMessage";
    //       设备安全
    API["rsa"] = "rsa";
    API["isSystemRoot"] = "isSystemRoot";
    //   蓝牙
    //      传统蓝牙
    API["openBluetoothAdapter"] = "openBluetoothAdapter";
    API["closeBluetoothAdapter"] = "closeBluetoothAdapter";
    API["getBluetoothAdapterState"] = "getBluetoothAdapterState";
    API["startBluetoothDevicesDiscovery"] = "startBluetoothDevicesDiscovery";
    API["stopBluetoothDevicesDiscovery"] = "stopBluetoothDevicesDiscovery";
    API["getBluetoothDevices"] = "getBluetoothDevices";
    API["getConnectedBluetoothDevices"] = "getConnectedBluetoothDevices";
    API["onBluetoothAdapterStateChange"] = "onBluetoothAdapterStateChange";
    API["offBluetoothAdapterStateChange"] = "offBluetoothAdapterStateChange";
    API["onBluetoothDeviceFound"] = "onBluetoothDeviceFound";
    API["offBluetoothDeviceFound"] = "offBluetoothDeviceFound";
    //       低功耗蓝牙
    API["connectBLEDevice"] = "connectBLEDevice";
    API["disconnectBLEDevice"] = "disconnectBLEDevice";
    API["getBLEDeviceServices"] = "getBLEDeviceServices";
    API["getBLEDeviceCharacteristics"] = "getBLEDeviceCharacteristics";
    API["notifyBLECharacteristicValueChange"] = "notifyBLECharacteristicValueChange";
    API["onBLECharacteristicValueChange"] = "onBLECharacteristicValueChange";
    API["offBLECharacteristicValueChange"] = "offBLECharacteristicValueChange";
    API["onBLEConnectionStateChanged"] = "onBLEConnectionStateChanged";
    API["offBLEConnectionStateChanged"] = "offBLEConnectionStateChanged";
    API["writeBLECharacteristicValue"] = "writeBLECharacteristicValue";
    API["readBLECharacteristicValue"] = "readBLECharacteristicValue";
    API["cancelBluetoothPair"] = "cancelBluetoothPair";
    API["getBluetoothPairs"] = "getBluetoothPairs";
    //      蓝牙从模式
    API["openBLEPeripheral"] = "openBLEPeripheral";
    API["closeBLEPeripheral"] = "closeBLEPeripheral";
    API["addPeripheralService"] = "addPeripheralService";
    API["removePeripheralService"] = "removePeripheralService";
    API["updateCharacteristic"] = "updateCharacteristic";
    API["startBLEAdvertising"] = "startBLEAdvertising";
    API["stopBLEAdvertising"] = "stopBLEAdvertising";
    API["onDeviceConnected"] = "onDeviceConnected";
    API["offDeviceConnected"] = "offDeviceConnected";
    API["onDeviceDisconnected"] = "onDeviceDisconnected";
    API["offDeviceDisconnected"] = "offDeviceDisconnected";
    API["onCharacteristicRead"] = "onCharacteristicRead";
    API["offCharacteristicRead"] = "offCharacteristicRead";
    API["onCharacteristicWrite"] = "onCharacteristicWrite";
    API["offCharacteristicWrite"] = "offCharacteristicWrite";
    API["onCharacteristicDidSubscribe"] = "onCharacteristicDidSubscribe";
    API["offCharacteristicDidSubscribe"] = "offCharacteristicDidSubscribe";
    API["onCharacteristicDidUnsubscribe"] = "onCharacteristicDidUnsubscribe";
    API["offCharacteristicDidUnsubscribe"] = "offCharacteristicDidUnsubscribe";
    //      iBeacon
    API["startBeaconDiscovery"] = "startBeaconDiscovery";
    API["stopBeaconDiscovery"] = "stopBeaconDiscovery";
    API["getBeacons"] = "getBeacons";
    API["onBeaconUpdate"] = "onBeaconUpdate";
    API["offBeaconUpdate"] = "offBeaconUpdate";
    API["onBeaconServiceChange"] = "onBeaconServiceChange";
    API["offBeaconServiceChange"] = "offBeaconServiceChange";
    // 位置
    //    用户操作定位
    API["chooseLocation"] = "chooseLocation";
    API["chooseCity"] = "chooseCity";
    API["onLocatedComplete"] = "onLocatedComplete";
    API["offLocatedComplete"] = "offLocatedComplete";
    API["openLocation"] = "openLocation";
    API["regionPicker"] = "regionPicker";
    //    接口操作定位
    API["getLocation"] = "getLocation";
    API["reGeoCode"] = "reGeoCode";
    API["geoCode"] = "geoCode";
    API["setLocatedCity"] = "setLocatedCity";
    //    持续定位
    API["onContinuousLocation"] = "onContinuousLocation";
    API["offContinuousLocation"] = "offContinuousLocation";
    API["startContinuousLocation"] = "startContinuousLocation";
    API["stopContinuousLocation"] = "stopContinuousLocation";
    //  网络请求
    //    Http 请求
    API["request"] = "request";
    API["httpRequest"] = "httpRequest";
    //    网关请求
    API["rpc"] = "rpc";
    API["sendMtop"] = "sendMtop";
    API["openRpc"] = "openRpc";
    //    Socket 请求
    API["connectSocket"] = "connectSocket";
    API["sendSocketMessage"] = "sendSocketMessage";
    API["closeSocket"] = "closeSocket";
    API["onSocketOpen"] = "onSocketOpen";
    API["offSocketOpen"] = "offSocketOpen";
    API["onSocketMessage"] = "onSocketMessage";
    API["offSocketMessage"] = "offSocketMessage";
    API["onSocketError"] = "onSocketError";
    API["offSocketError"] = "offSocketError";
    API["onSocketClose"] = "onSocketClose";
    API["offSocketClose"] = "offSocketClose";
    //    TCP/UDP
    API["createTCPSocket"] = "createTCPSocket";
    API["createUDPSocket"] = "createUDPSocket";
    //    上传下载
    API["uploadFile"] = "uploadFile";
    API["uploadFileToAliCloud"] = "uploadFileToAliCloud";
    API["downloadFile"] = "downloadFile";
    // 文件与存储
    //    键值对缓存
    API["setStorage"] = "setStorage";
    API["getStorage"] = "getStorage";
    API["removeStorage"] = "removeStorage";
    API["clearStorage"] = "clearStorage";
    API["getStorageInfo"] = "getStorageInfo";
    API["setStorageSync"] = "setStorageSync";
    API["getStorageSync"] = "getStorageSync";
    API["removeStorageSync"] = "removeStorageSync";
    API["clearStorageSync"] = "clearStorageSync";
    API["getStorageInfoSync"] = "getStorageInfoSync";
    //    文件存储
    API["chooseFileFromDisk"] = "chooseFileFromDisk";
    API["saveFile"] = "saveFile";
    API["removeSavedFile"] = "removeSavedFile";
    API["openDocument"] = "openDocument";
    API["getSavedFileList"] = "getSavedFileList";
    API["getSavedFileInfo"] = "getSavedFileInfo";
    API["getFileInfo"] = "getFileInfo";
    //    文件系统
    API["getFileSystemManager"] = "getFileSystemManager";
    // 性能计算
    API["createXnnManager"] = "createXnnManager";
    API["createWorker"] = "createWorker";
    // 多媒体
    //   音频
    //     中断事件
    API["onAudioInterruptionBegin"] = "onAudioInterruptionBegin";
    API["onAudioInterruptionEnd"] = "onAudioInterruptionEnd";
    API["offAudioInterruptionBegin"] = "offAudioInterruptionBegin";
    API["offAudioInterruptionEnd"] = "offAudioInterruptionEnd";
    //     后台播放(旧)
    API["playBackgroundAudio"] = "playBackgroundAudio";
    API["pauseBackgroundAudio"] = "pauseBackgroundAudio";
    API["stopBackgroundAudio"] = "stopBackgroundAudio";
    API["seekBackgroundAudio"] = "seekBackgroundAudio";
    API["getBackgroundAudioPlayerState"] = "getBackgroundAudioPlayerState";
    API["onBackgroundAudioPlay"] = "onBackgroundAudioPlay";
    API["offBackgroundAudioPlay"] = "offBackgroundAudioPlay";
    API["onBackgroundAudioPause"] = "onBackgroundAudioPause";
    API["offBackgroundAudioPause"] = "offBackgroundAudioPause";
    API["onBackgroundAudioStop"] = "onBackgroundAudioStop";
    API["offBackgroundAudioStop"] = "offBackgroundAudioStop";
    //     后台播放管理器(新)
    API["getBackgroundAudioManager"] = "getBackgroundAudioManager";
    //     播放(旧)
    API["createAudioContext"] = "createAudioContext";
    //     播放(新)
    API["createInnerAudioContext"] = "createInnerAudioContext";
    //   实时音视频
    API["createLivePlayerContext"] = "createLivePlayerContext";
    API["createLivePusherContext"] = "createLivePusherContext";
    //   语音
    //     录音(旧)
    API["startRecord"] = "startRecord";
    API["stopRecord"] = "stopRecord";
    API["cancelRecord"] = "cancelRecord";
    //     录音管理器(新)
    API["getRecorderManager"] = "getRecorderManager";
    API["getAvailableAudioSources"] = "getAvailableAudioSources";
    //     播放语音
    API["playVoice"] = "playVoice";
    API["pauseVoice"] = "pauseVoice";
    API["resumeVoice"] = "resumeVoice";
    API["stopVoice"] = "stopVoice";
    //     TTS
    API["textToSpeech"] = "textToSpeech";
    //     语音识别
    API["isSpeechRecognizeAvailable"] = "isSpeechRecognizeAvailable";
    API["startSpeechRecognize"] = "startSpeechRecognize";
    API["stopSpeechRecognize"] = "stopSpeechRecognize";
    API["onSpeechRecognizeResult"] = "onSpeechRecognizeResult";
    API["offSpeechRecognizeResult"] = "offSpeechRecognizeResult";
    API["onSpeechRecognizeError"] = "onSpeechRecognizeError";
    API["offSpeechRecognizeError"] = "offSpeechRecognizeError";
    //   增强现实(AR)
    API["createARContext"] = "createARContext";
    API["ARScan"] = "ARScan";
    API["ap$ARScan"] = "ap.ARScan";
    //   图片
    API["chooseImage"] = "chooseImage";
    API["saveImageToPhotosAlbum"] = "saveImageToPhotosAlbum";
    API["previewImage"] = "previewImage";
    API["saveImage"] = "saveImage";
    API["getImageInfo"] = "getImageInfo";
    API["compressImage"] = "compressImage";
    API["generateImageFromCode"] = "generateImageFromCode";
    //   视频
    API["saveVideoToPhotosAlbum"] = "saveVideoToPhotosAlbum";
    API["chooseVideo"] = "chooseVideo";
    API["createVideoContext"] = "createVideoContext";
    //   地图
    API["createMapContext"] = "createMapContext";
    API["calculateRoute"] = "calculateRoute";
    //   相机
    API["ap$xnnOcr"] = "ap.xnnOcr";
    API["scan"] = "scan";
    API["ocr"] = "ocr";
    API["ap$ocr"] = "ap.ocr";
    API["createCameraContext"] = "createCameraContext";
    API["createAICameraContext"] = "createAICameraContext";
    // 留存
    //    分享
    API["hideShareMenu"] = "hideShareMenu";
    API["showSharePanel"] = "showSharePanel";
    //    收藏 (inside 没有)
    API["isFavorite"] = "isFavorite";
    API["addToFavorite"] = "addToFavorite";
    API["removeFromFavorite"] = "removeFromFavorite";
    API["onFavorite"] = "onFavorite";
    API["offFavorite"] = "offFavorite";
    API["onInternalFavorite"] = "onInternalFavorite";
    API["offInternalFavorite"] = "offInternalFavorite";
    API["isCollected"] = "isCollected";
    API["canFavorite"] = "canFavorite";
    API["hideFavoriteMenu"] = "hideFavoriteMenu";
    API["hideAllFavoriteMenu"] = "hideAllFavoriteMenu";
    // 性能
    //   首屏快照
    API["enableSnapshot"] = "enableSnapshot";
    API["reportSnapshot"] = "reportSnapshot";
    API["saveSnapshot"] = "saveSnapshot";
    // 开放接口
    //      用户信息
    API["getAuthCode"] = "getAuthCode";
    API["ap$getAuthCode"] = "ap.getAuthCode";
    API["authorize"] = "authorize";
    API["ap$authorize"] = "ap.authorize";
    API["getAuthUserInfo"] = "getAuthUserInfo";
    API["ap$getAuthUserInfo"] = "ap.getAuthUserInfo";
    API["getUserInfo"] = "getUserInfo";
    API["ap$getUserInfo"] = "ap.getUserInfo";
    API["getOpenUserInfo"] = "getOpenUserInfo";
    API["getOpenUserData"] = "getOpenUserData";
    API["getPhoneNumber"] = "getPhoneNumber";
    API["getIDNumber"] = "getIDNumber";
    API["getVisitToken"] = "getVisitToken";
    //      设备授权
    API["showAuthGuide"] = "showAuthGuide";
    API["getAuthStatus"] = "getAuthStatus";
    //   插件
    API["loadPlugin"] = "loadPlugin";
    API["getParentAppId"] = "getParentAppId";
    API["getParentAppIdSync"] = "getParentAppIdSync";
    //   模板小程序
    API["getExtConfig"] = "getExtConfig";
    API["getExtConfigSync"] = "getExtConfigSync";
    //    小服务
    API["navigateToMiniService"] = "navigateToMiniService";
    API["startMiniService"] = "startMiniService";
    API["navigateBackFromMiniService"] = "navigateBackFromMiniService";
    API["navigateToMiniProgram"] = "navigateToMiniProgram";
    API["navigateBackMiniProgram"] = "navigateBackMiniProgram";
    //    H5
    API["ap$navigateToAlipayPage"] = "ap.navigateToAlipayPage";
    //   应用跳转
    API["openOtherApplication"] = "openOtherApplication";
    API["ap$openOtherApplication"] = "ap.openOtherApplication";
    // 智能硬件(IoT)
    //    消息
    API["onIotRawMessage"] = "onIotRawMessage";
    API["offIotRawMessage"] = "offIotRawMessage";
    API["onIotNotify"] = "onIotNotify";
    API["offIotNotify"] = "offIotNotify";
    //   请求
    API["sendLanHttpRequest"] = "sendLanHttpRequest";
    //   服务
    API["iotCreateService"] = "iotCreateService";
    API["ap$iotCreateService"] = "ap.iotCreateService";
    API["iotCheckService"] = "iotCheckService";
    API["ap$iotCheckService"] = "ap.iotCheckService";
    //  业务能力
    //    运动
    API["getRunData"] = "getRunData";
    //    内容安全
    API["textRiskIdentification"] = "textRiskIdentification";
    API["ap$textRiskIdentification"] = "ap.textRiskIdentification";
    API["ap$imgRisk"] = "ap.imgRisk";
    API["ap$imgRiskCallback"] = "ap.imgRiskCallback";
    //    淘宝
    API["openTaobao"] = "openTaobao";
    API["ap$openTaobao"] = "ap.openTaobao";
    API["getTBCode"] = "getTBCode";
    API["setTBSessionInfo"] = "setTBSessionInfo";
    API["getTBSessionInfo"] = "getTBSessionInfo";
    //   支付及资金
    API["tradePay"] = "tradePay";
    API["ap$tradePay"] = "ap.tradePay";
    API["paySignCenter"] = "paySignCenter";
    API["ap$paySignCenter"] = "ap.paySignCenter";
    // getConfigService = 'getConfigService', // my. & my.ap. & configService.getConfig I mean only Alipay
    //   云客服
    API["openCustomerService"] = "openCustomerService";
    API["ap$openCustomerService"] = "ap.openCustomerService";
    //   聊天会话
    API["ap$openChatWindow"] = "ap.openChatWindow";
    API["openChatWindow"] = "openChatWindow";
    //   券详情
    API["navigateToCouponDetail"] = "navigateToCouponDetail";
    API["ap$navigateToCouponDetail"] = "ap.navigateToCouponDetail";
    //   车主服务
    API["openCarService"] = "openCarService";
    API["ap$openCarService"] = "ap.openCarService";
    //   财富
    API["ap$navigateToFinance"] = "ap.navigateToFinance";
    // 生活号
    API["ap$cancelFollowLifestyle"] = "ap.cancelFollowLifestyle";
    API["ap$isFollowLifestyle"] = "ap.isFollowLifestyle";
    //    卡包
    API["openVoucherList"] = "openVoucherList";
    API["ap$openVoucherList"] = "ap.openVoucherList";
    API["openTicketList"] = "openTicketList";
    API["ap$openTicketList"] = "ap.openTicketList";
    API["openVoucherDetail"] = "openVoucherDetail";
    API["ap$openVoucherDetail"] = "ap.openVoucherDetail";
    API["openTicketDetail"] = "openTicketDetail";
    API["ap$openTicketDetail"] = "ap.openTicketDetail";
    API["openMerchantVoucherList"] = "openMerchantVoucherList";
    API["ap$openMerchantVoucherList"] = "ap.openMerchantVoucherList";
    API["openMerchantTicketList"] = "openMerchantTicketList";
    API["ap$openMerchantTicketList"] = "ap.openMerchantTicketList";
    API["openKBVoucherDetail"] = "openKBVoucherDetail";
    API["ap$openKBVoucherDetail"] = "ap.openKBVoucherDetail";
    //   商户会员卡
    API["addCardAuth"] = "addCardAuth";
    API["ap$addCardAuth"] = "ap.addCardAuth";
    API["openCardList"] = "openCardList";
    API["ap$openCardList"] = "ap.openCardList";
    API["openCardDetail"] = "openCardDetail";
    API["ap$openCardDetail"] = "ap.openCardDetail";
    API["openMerchantCardList"] = "openMerchantCardList";
    API["ap$openMerchantCardList"] = "ap.openMerchantCardList";
    //     营销反作弊
    API["createRDSContext"] = "createRDSContext";
    API["ap$preventCheat"] = "ap.preventCheat";
    //     先享后付保障
    API["ap$nsf"] = "ap.nsf";
    //       核身
    API["startAPVerify"] = "startAPVerify";
    API["ap$startAPVerify"] = "ap.startAPVerify";
    API["ap$getVerifyEnvData"] = "ap.getVerifyEnvData";
    API["ap$startVerifyIdentity"] = "ap.startVerifyIdentity";
    API["ap$faceVerify"] = "ap.faceVerify";
    API["ap$zimIdentity"] = "ap.zimIdentity";
    //    发票
    API["chooseInvoiceTitle"] = "chooseInvoiceTitle";
    API["ap$chooseInvoiceTitle"] = "ap.chooseInvoiceTitle";
    //    收货地址
    // 支付宝多端可以
    API["chooseAddress"] = "chooseAddress";
    API["ap$chooseAddress"] = "ap.chooseAddress";
    API["getAddress"] = "getAddress";
    API["ap$getAddress"] = "ap.getAddress";
    //    芝麻信用 (inside 还没，可以有)
    API["zmCreditBorrow"] = "zmCreditBorrow";
    API["ap$zmCreditBorrow"] = "ap.zmCreditBorrow";
    API["zmRentTransition"] = "zmRentTransition";
    API["ap$zmRentTransition"] = "ap.zmRentTransition";
    API["zmFreeDeposit"] = "zmFreeDeposit";
    API["ap$zmFreeDeposit"] = "ap.zmFreeDeposit";
    API["startZMCreditRent"] = "startZMCreditRent";
    API["ap$startZMCreditRent"] = "ap.startZMCreditRent";
    API["startZMVerify"] = "startZMVerify";
    API["ap$startZMVerify"] = "ap.startZMVerify";
    // generateImageFromCode = "generateImageFromCode",
    // ap$subscribeMsgbox = 'ap.subscribeMsgbox',
    // ap$getAdTrackClues = 'ap.getAdTrackClues',
    // donateInBrowser = "donateInBrowser",
})(API = exports.API || (exports.API = {}));
