// @alipay/mini-api-types/lib/public/apis/storage/fs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Encoding;
(function (Encoding) {
    Encoding["ascii"] = "ascii";
    Encoding["base64"] = "base64";
    Encoding["hex"] = "hex";
    Encoding["binary"] = "binary";
    Encoding["ucs2"] = "ucs2";
    Encoding["ucs-2"] = "ucs-2";
    Encoding["utf16le"] = "utf16le";
    Encoding["utf-16le"] = "utf-16le";
    Encoding["utf-8"] = "utf-8";
    Encoding["utf8"] = "utf8";
    Encoding["latin1"] = "latin1";
})(Encoding || (Encoding = {}));
var EFileSystemErrorCode;
(function (EFileSystemErrorCode) {
    /**
     * @description 指定文件/目录不存在。
     */
    EFileSystemErrorCode[EFileSystemErrorCode["_10022"] = 10022] = "_10022";
    /**
     * @description 指定路径已经是一个已存在的目录。
     */
    EFileSystemErrorCode[EFileSystemErrorCode["_10023"] = 10023] = "_10023";
    /**
     * @description 指定的路径没有读或者写的权限。
     */
    EFileSystemErrorCode[EFileSystemErrorCode["_10024"] = 10024] = "_10024";
    /**
     * @description 指定路径是一个已经存在的目录。
     */
    EFileSystemErrorCode[EFileSystemErrorCode["_10025"] = 10025] = "_10025";
    /**
     * @description 写入文件单个超过10M或者写入文件夹超过50M。
     */
    EFileSystemErrorCode[EFileSystemErrorCode["_10028"] = 10028] = "_10028";
    /**
     * @description 删除文件失败。
     */
    EFileSystemErrorCode[EFileSystemErrorCode["_15"] = 15] = "_15";
    /**
     * @description 未知错误。
     */
    EFileSystemErrorCode[EFileSystemErrorCode["_3"] = 3] = "_3";
})(EFileSystemErrorCode || (EFileSystemErrorCode = {}));
