// @alipay/mini-api-types/lib/public/type/alipay.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIScope = void 0;
var APIScope;
(function (APIScope) {
    // 这些全都独立以 支付宝 接口来写，不然要乱掉了
    // TODO需要设计一个 namespace 的类型推断机制
    // 支持 namespace + namespaceAndRoot 两个模式
    var Alipay;
    (function (Alipay) {
        Alipay["aproot$getAuthCode"] = "getAuthCode";
        Alipay["aproot$getAuthUserInfo"] = "getAuthUserInfo";
        Alipay["aproot$authorize"] = "authorize";
        // 一定是支付宝自有的
        Alipay["aproot$getUserInfo"] = "getUserInfo";
        Alipay["aproot$getConfigService"] = "getConfigService";
        Alipay["aproot$ocr"] = "ocr";
        Alipay["aproot$openCustomerService"] = "openCustomerService";
        Alipay["aproot$startAPVerify"] = "startAPVerify";
        Alipay["aproot$chooseAddress"] = "chooseAddress";
        Alipay["aproot$getAddress"] = "getAddress";
        Alipay["aproot$openTaobao"] = "openTaobao";
        Alipay["aproot$navigateToCouponDetail"] = "navigateToCouponDetail";
        Alipay["aproot$chooseInvoiceTitle"] = "chooseInvoiceTitle";
        Alipay["aproot$openCarService"] = "openCarService";
        Alipay["aproot$openChatWindow"] = "openChatWindow";
        Alipay["aproot$zmCreditBorrow"] = "zmCreditBorrow";
        Alipay["aproot$zmRentTransition"] = "zmRentTransition";
        Alipay["aproot$zmFreeDeposit"] = "zmFreeDeposit";
        Alipay["aproot$startZMCreditRent"] = "startZMCreditRent";
        Alipay["aproot$startZMVerify"] = "startZMVerify";
        Alipay["aproot$iotCreateService"] = "iotCreateService";
        Alipay["aproot$iotCheckService"] = "iotCheckService";
        // mdns
        Alipay["aproot$onLocalServiceResolveFail"] = "onLocalServiceResolveFail";
        Alipay["aproot$onLocalServiceLost"] = "onLocalServiceLost";
        Alipay["aproot$onLocalServiceFound"] = "onLocalServiceFound";
        Alipay["aproot$onLocalServiceDiscoveryStop"] = "onLocalServiceDiscoveryStop";
        Alipay["aproot$offLocalServiceResolveFail"] = "offLocalServiceResolveFail";
        Alipay["aproot$offLocalServiceLost"] = "offLocalServiceLost";
        Alipay["aproot$offLocalServiceFound"] = "offLocalServiceFound";
        Alipay["aproot$offLocalServiceDiscoveryStop"] = "offLocalServiceDiscoveryStop";
        Alipay["aproot$stopLocalServiceDiscovery"] = "stopLocalServiceDiscovery";
        Alipay["aproot$startLocalServiceDiscovery"] = "startLocalServiceDiscovery";
        // 淘宝都有
        Alipay["aproot$tradePay"] = "tradePay";
        // 支付宝 inside 看心情有
        Alipay["aproot$textRiskIdentification"] = "textRiskIdentification";
        Alipay["aproot$paySignCenter"] = "paySignCenter";
        // 支付宝系 inside 有
        Alipay["aproot$openSetting"] = "openSetting";
        Alipay["aproot$getSetting"] = "getSetting";
        Alipay["aproot$openOtherApplication"] = "openOtherApplication";
        // 一定是私有，虽然有 my.topNS 支持
        Alipay["aproot$ARScan"] = "ARScan";
        Alipay["aproot$addCardAuth"] = "addCardAuth";
        Alipay["aproot$openCardList"] = "openCardList";
        Alipay["aproot$openVoucherList"] = "openVoucherList";
        Alipay["aproot$openTicketList"] = "openTicketList";
        Alipay["aproot$openCardDetail"] = "openCardDetail";
        Alipay["aproot$openVoucherDetail"] = "openVoucherDetail";
        Alipay["aproot$openTicketDetail"] = "openTicketDetail";
        Alipay["aproot$openMerchantCardList"] = "openMerchantCardList";
        Alipay["aproot$openMerchantVoucherList"] = "openMerchantVoucherList";
        Alipay["aproot$openMerchantTicketList"] = "openMerchantTicketList";
        Alipay["aproot$openKBVoucherDetail"] = "openKBVoucherDetail";
        Alipay["ap$preventCheat"] = "preventCheat";
        Alipay["ap$nsf"] = "nsf";
        Alipay["ap$imgRisk"] = "imgRisk";
        Alipay["ap$imgRiskCallback"] = "imgRiskCallback";
        Alipay["ap$navigateToFinance"] = "navigateToFinance";
        Alipay["ap$navigateToAlipayPage"] = "navigateToAlipayPage";
        Alipay["ap$cancelFollowLifestyle"] = "cancelFollowLifestyle";
        Alipay["ap$isFollowLifestyle"] = "isFollowLifestyle";
        Alipay["ap$subscribeMsgbox"] = "subscribeMsgbox";
        Alipay["ap$xnnOcr"] = "xnnOcr";
        Alipay["ap$updateAlipayClient"] = "updateAlipayClient";
        Alipay["ap$getVerifyEnvData"] = "getVerifyEnvData";
        Alipay["ap$startVerifyIdentity"] = "startVerifyIdentity";
        Alipay["ap$getAdTrackClues"] = "getAdTrackClues";
        Alipay["ap$faceVerify"] = "faceVerify";
        Alipay["ap$zimIdentity"] = "zimIdentity";
    })(Alipay = APIScope.Alipay || (APIScope.Alipay = {}));
})(APIScope = exports.APIScope || (exports.APIScope = {}));
