// @alipay/mini-api-types/lib/public/apis/open-biz/card.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var EAddCardAuthErrorCode;
(function (EAddCardAuthErrorCode) {
    /**
     * @description 用户取消。
     */
    EAddCardAuthErrorCode["JSAPI_SERVICE_TERMINATED"] = "JSAPI_SERVICE_TERMINATED";
    /**
     * @description URL 为空或非法参数。
     */
    EAddCardAuthErrorCode["JSAPI_PARAM_INVALID"] = "JSAPI_PARAM_INVALID";
    /**
     * @description 系统错误。
     */
    EAddCardAuthErrorCode["JSAPI_SYSTEM_ERROR"] = "JSAPI_SYSTEM_ERROR";
})(EAddCardAuthErrorCode || (EAddCardAuthErrorCode = {}));
/**
 * @description IoT 小程序错误码
 */
var EAddCardAuthIoTErrorCode;
(function (EAddCardAuthIoTErrorCode) {
    /**
     * @description 非法空参数，通过 alipay.marketing.card.activateurl.apply 接口生成的领卡链接，剔除 callback 参数，并对领卡链接做 urlDecode 处理。
     */
    EAddCardAuthIoTErrorCode["ILLEGAL_NULL_ARGUMENT"] = "ILLEGAL_NULL_ARGUMENT";
    /**
     * @description 非法的开卡链接，通过 alipay.marketing.card.activateurl.apply 接口生成的领卡链接，剔除 callback 参数，并对领卡链接做 urlDecode 处理。
     */
    EAddCardAuthIoTErrorCode["INVALID_APPLY_CARD_LINK"] = "INVALID_APPLY_CARD_LINK";
    /**
     * @description 开放表单配置信息不存在，调用 alipay.marketing.card.formtemplate.set 接口配置表单信息。
     */
    EAddCardAuthIoTErrorCode["OPEN_FORM_TEMPLATE_NOT_EXIST"] = "OPEN_FORM_TEMPLATE_NOT_EXIST";
    /**
     * @description 会员已有会员卡，如需测试重新领取，可到支付宝卡包中删除会员卡，重新领取。
     */
    EAddCardAuthIoTErrorCode["ALREADY_ACTIVE_CARD"] = "ALREADY_ACTIVE_CARD";
    /**
     * @description 表单字段商家无权限，后台申请对应缺失字段的权限，参考《商户自助申请用户信息》。
     */
    EAddCardAuthIoTErrorCode["OPEN_FORM_FIELD_NOT_PERMITTED"] = "OPEN_FORM_FIELD_NOT_PERMITTED";
    /**
     * @description 当前表单用户信息缺失。
     * - 会员卡表单设置姓名、手机号等字段，但是会员卡应用下没有这些字段的权限。后台申请对应缺失字段的权限，请参考[《商户自助申请用户信息》](https://opendocs.alipay.com/mini/multi-platform/dz1u63)。
     * - 支付宝缺少此用户的表单信息，无法获取到信息。建议用户进行实名认证，补充缺少的姓名、手机号等信息。
     */
    EAddCardAuthIoTErrorCode["OPEN_FORM_USER_LACK_INFO"] = "OPEN_FORM_USER_LACK_INFO";
    /**
     * @description 系统异常，建议稍后重试。
     */
    EAddCardAuthIoTErrorCode["SYSTEM_ERROR"] = "SYSTEM_ERROR";
})(EAddCardAuthIoTErrorCode || (EAddCardAuthIoTErrorCode = {}));
