// @alipay/mini-api-types/lib/public/apis/open-biz/credit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ERentUnit;
(function (ERentUnit) {
    /**
     * @description 元/天
     */
    ERentUnit["DAY_YUAN"] = "DAY_YUAN";
    /**
     * @description 元/小时
     */
    ERentUnit["HOUR_YUAN"] = "HOUR_YUAN";
    /**
     * @description 元
     */
    ERentUnit["YUAN"] = "YUAN";
    /**
     * @description  元/次
     */
    ERentUnit["YUAN_ONCE"] = "YUAN_ONCE";
})(ERentUnit || (ERentUnit = {}));
var EDepositState;
(function (EDepositState) {
    /**
     * @description 支持
     */
    EDepositState["Y"] = "Y";
    /**
     * @description 不支持
     */
    EDepositState["N"] = "N";
})(EDepositState || (EDepositState = {}));
var EInvokeType;
(function (EInvokeType) {
    /**
     * @description 回跳至小程序地址
     */
    EInvokeType["TINYAPP"] = "TINYAPP";
    /**
     * @description 支付宝服务窗
     */
    EInvokeType["WINDOWS"] = "WINDOWS";
})(EInvokeType || (EInvokeType = {}));
var ERentSettleType;
(function (ERentSettleType) {
    /**
     * @description 表示商户自行结算，信用借还不提供租金支付能力
     */
    ERentSettleType["merchant"] = "merchant";
    /**
     * @description 表示使用支付宝支付功能，给用户提供租金代扣及赔偿金支付能力；
     */
    ERentSettleType["alipay"] = "alipay";
})(ERentSettleType || (ERentSettleType = {}));
var ESpuName;
(function (ESpuName) {
    ESpuName["power_bank"] = "power_bank";
    ESpuName["umbrella"] = "umbrella";
})(ESpuName || (ESpuName = {}));
var EZmCreditBorrowErrorCode;
(function (EZmCreditBorrowErrorCode) {
    /**
     * @description 参数有误，请确认参数格式。
     */
    EZmCreditBorrowErrorCode["INVALID_PARAMETER"] = "INVALID_PARAMETER";
    /**
     * @description 商户对产品未签约，请联系业务与芝麻签约信用借还产品。
     */
    EZmCreditBorrowErrorCode["ARRANGEMENT_NOT_EXIST"] = "ARRANGEMENT_NOT_EXIST";
    /**
     * @description 目前不支持押金模式，请与芝麻业务确认。
     */
    EZmCreditBorrowErrorCode["DEPOSIT_CLOSED"] = "DEPOSIT_CLOSED";
    /**
     * @description 该物品已经借用，请先归还后再借。
     */
    EZmCreditBorrowErrorCode["GOODS_HAS_BORROW"] = "GOODS_HAS_BORROW";
    /**
     * @description 该用户预授权冻结押金失败，请确认支付宝账号中是否有足够的金额。
     */
    EZmCreditBorrowErrorCode["PREAUTH_FREEZE_ERROR"] = "PREAUTH_FREEZE_ERROR";
    /**
     * @description 签约代扣协议错误，请联系支付宝客服。
     */
    EZmCreditBorrowErrorCode["WITHHOLD_CONTRACT_ERROR"] = "WITHHOLD_CONTRACT_ERROR";
    /**
     * @description 芝麻会员开通失败，请确保支付宝会员是否正常。
     */
    EZmCreditBorrowErrorCode["ZM_ACCOUT_REGISTER_FAIL"] = "ZM_ACCOUT_REGISTER_FAIL";
    /**
     * @description 外部订单号重复，请更换外部订单号。
     */
    EZmCreditBorrowErrorCode["OUT_ORDER_NO_INVALID"] = "OUT_ORDER_NO_INVALID";
    /**
     * @description 该支付宝账号没有登录，请登录支付宝。
     */
    EZmCreditBorrowErrorCode["USER_NOT_LOGON"] = "USER_NOT_LOGON";
    /**
     * @description 信息含敏感内容，请检查入参信息是否包含敏感内容。
     */
    EZmCreditBorrowErrorCode["INFOSEC_VERIFY_FAILED"] = "INFOSEC_VERIFY_FAILED";
})(EZmCreditBorrowErrorCode || (EZmCreditBorrowErrorCode = {}));
var EOrderStatus;
(function (EOrderStatus) {
    /**
     * @description 下单成功
     */
    EOrderStatus["SUCCESS"] = "SUCCESS";
    /**
     * @description 下单失败
     */
    EOrderStatus["FAIL"] = "FAIL";
    /**
     * @description 处理中
     */
    EOrderStatus["UNKOWN"] = "UNKOWN";
})(EOrderStatus || (EOrderStatus = {}));
var EZmRentTransitionErrorCode;
(function (EZmRentTransitionErrorCode) {
    /**
     * @description 系统错误，稍后再试。
     */
    EZmRentTransitionErrorCode["SYSTEM_FAILURE"] = "SYSTEM_FAILURE";
    /**
     * @description 订单不存在。
     */
    EZmRentTransitionErrorCode["ORDER_NOT_EXISTS"] = "ORDER_NOT_EXISTS";
    /**
     * @description 验签错误。
     */
    EZmRentTransitionErrorCode["SIGN_ERROR"] = "SIGN_ERROR";
    /**
     * @description 签约代扣协议失败。
     */
    EZmRentTransitionErrorCode["WITHHOLDING_FAILED"] = "WITHHOLDING_FAILED";
    /**
     * @description 该用户预授权冻结资金失败，请确认支付宝账号中是否有足够的金额。
     */
    EZmRentTransitionErrorCode["PREAUTH_FREEZE_ERROR"] = "PREAUTH_FREEZE_ERROR";
    /**
     * @description 订单确认失败。
     */
    EZmRentTransitionErrorCode["ORDER_CONFIRM_ERROR"] = "ORDER_CONFIRM_ERROR";
})(EZmRentTransitionErrorCode || (EZmRentTransitionErrorCode = {}));
var EZmFreeDepositEnv;
(function (EZmFreeDepositEnv) {
    /**
     * @description 生产
     */
    EZmFreeDepositEnv["prod"] = "prod";
    /**
     * @description 测试
     */
    EZmFreeDepositEnv["stable"] = "stable";
})(EZmFreeDepositEnv || (EZmFreeDepositEnv = {}));
var EZmFreeDepositStatusCode;
(function (EZmFreeDepositStatusCode) {
    /**
     * @description 有授信额度
     */
    EZmFreeDepositStatusCode["HAS_LIMIT"] = "HAS_LIMIT";
    /**
     * @description 无授信额度
     */
    EZmFreeDepositStatusCode["NO_LIMIT"] = "NO_LIMIT";
})(EZmFreeDepositStatusCode || (EZmFreeDepositStatusCode = {}));
