// @alipay/mini-api-types/lib/public/apis/open-biz/verify.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var EFaceVerifyRetCode;
(function (EFaceVerifyRetCode) {
    /**
     * @description 刷脸可用，查询faceRetCode查看是否成功刷脸
     */
    EFaceVerifyRetCode["OK_SUCCESS"] = "OK_SUCCESS";
    /**
     * @description 系统错误
     */
    EFaceVerifyRetCode["SYSTEM_ERROR"] = "SYSTEM_ERROR";
    /**
     * @description 用户不可用刷脸
     */
    EFaceVerifyRetCode["ACCOUNT_SERVICE_SUSPEND"] = "ACCOUNT_SERVICE_SUSPEND";
    /**
     * @description 刷脸失败次数过多，请您稍后再试
     */
    EFaceVerifyRetCode["HIGH_RISK"] = "HIGH_RISK";
    /**
     * @description 设备不支持刷脸
     */
    EFaceVerifyRetCode["DEVICE_NOT_SUPPORT"] = "DEVICE_NOT_SUPPORT";
    /**
     * @description 手机操作系统不支持刷脸
     */
    EFaceVerifyRetCode["OS_NOT_SUPPORT"] = "OS_NOT_SUPPORT";
    /**
     * @description SDK版本不支持
     */
    EFaceVerifyRetCode["SDKVERSION_NOT_SUPPORT"] = "SDKVERSION_NOT_SUPPORT";
    /**
     * @description 刷脸资源未准备好
     */
    EFaceVerifyRetCode["ASSET_NOT_AVAILABLE"] = "ASSET_NOT_AVAILABLE";
    /**
     * @description 比对源不可用
     */
    EFaceVerifyRetCode["UNABLE_GET_IMAGE"] = "UNABLE_GET_IMAGE";
    /**
     * @description 算法不支持
     */
    EFaceVerifyRetCode["ALG_NOT_SUPPORT"] = "ALG_NOT_SUPPORT";
    /**
     * @description 非实名认证用户
     */
    EFaceVerifyRetCode["NOT_REAL_NAME"] = "NOT_REAL_NAME";
})(EFaceVerifyRetCode || (EFaceVerifyRetCode = {}));
var EFaceVerifyFaceRetCode;
(function (EFaceVerifyFaceRetCode) {
    /**
     * @description 代表刷脸成功
     */
    EFaceVerifyFaceRetCode[EFaceVerifyFaceRetCode["_1000"] = 1000] = "_1000";
    /**
     * @description 表示系统错误
     */
    EFaceVerifyFaceRetCode[EFaceVerifyFaceRetCode["_1001"] = 1001] = "_1001";
    /**
     * @description 表示验证中断
     */
    EFaceVerifyFaceRetCode[EFaceVerifyFaceRetCode["_1003"] = 1003] = "_1003";
    /**
     * @description 表示网络错误
     */
    EFaceVerifyFaceRetCode[EFaceVerifyFaceRetCode["_2002"] = 2002] = "_2002";
    /**
     * @description 表示刷脸失败
     */
    EFaceVerifyFaceRetCode[EFaceVerifyFaceRetCode["_2006"] = 2006] = "_2006";
})(EFaceVerifyFaceRetCode || (EFaceVerifyFaceRetCode = {}));
var EVerifyType;
(function (EVerifyType) {
    /**
     * @description 标准核身
     */
    EVerifyType["standard"] = "standard";
    /**
     * @description 快捷核身1（核身SDK直接初始化）
     */
    EVerifyType["verify_init"] = "verify_init";
    /**
     * @description 快捷核身2（业务服务端初始化）
     */
    EVerifyType["verify_module\uFE0E"] = "verify_module\uFE0E";
})(EVerifyType || (EVerifyType = {}));
var EVerifyCode;
(function (EVerifyCode) {
    /**
     * @description [人脸] 人脸比对不是同一个人
     */
    EVerifyCode["NOT_SAME_PERSON"] = "NOT_SAME_PERSON";
    /**
     * @description [人脸] 人脸无法获取图片
     */
    EVerifyCode["UNABLE_GET_IMAGE"] = "UNABLE_GET_IMAGE";
    /**
     * @description [人脸] 人脸系统异常
     */
    EVerifyCode["SYSTEM_ERROR"] = "SYSTEM_ERROR";
    /**
     * @description [支付密码] 支付密码被锁定，用户不进入 找回密码流程
     */
    EVerifyCode["PPW_LOCK_USER_CANCEL"] = "PPW_LOCK_USER_CANCEL";
})(EVerifyCode || (EVerifyCode = {}));
