// @alipay/mini-api-types/lib/public/apis/open/device-auth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var EAuthType4Android;
(function (EAuthType4Android) {
    /**
     * @description 后台保活权限 (Android)
     */
    EAuthType4Android["BACKGROUNDER"] = "BACKGROUNDER";
    /**
     * @description 桌面快捷权限 (Android)
     */
    EAuthType4Android["SHORTCUT"] = "SHORTCUT";
    /**
     * @description Push 通知栏权限 (Android)
     */
    EAuthType4Android["NOTIFICATION"] = "NOTIFICATION";
    /**
     * @description 自启动权限 (Android)
     */
    EAuthType4Android["SELFSTARTING"] = "SELFSTARTING";
})(EAuthType4Android || (EAuthType4Android = {}));
var EAuthType4IOS;
(function (EAuthType4IOS) {
    /**
     * @description 麦克风权限 (iOS)
     */
    EAuthType4IOS["MICROPHONE"] = "MICROPHONE";
    /**
     * @description 通讯录权限 (iOS)
     */
    EAuthType4IOS["ADDRESSBOOK"] = "ADDRESSBOOK";
    /**
     * @description 相机权限 (iOS)
     */
    EAuthType4IOS["CAMERA"] = "CAMERA";
    /**
     * @description 照片权限 (iOS)
     */
    EAuthType4IOS["PHOTO"] = "PHOTO";
    /**
     * @description LBS 总开关 (iOS)
     */
    EAuthType4IOS["LBSSERVICE"] = "LBSSERVICE";
    /**
     * @description LBS 开关（应用级别） (iOS)
     */
    EAuthType4IOS["LBS"] = "LBS";
})(EAuthType4IOS || (EAuthType4IOS = {}));
