// @alipay/mini-api-types/lib/public/apis/open/navigation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var envVersion;
(function (envVersion) {
    /**
     * @description 开发版
     */
    envVersion["develop"] = "develop";
    /**
     * @description 体验版
     */
    envVersion["trial"] = "trial";
    /**
     * @description 发布版
     */
    envVersion["release"] = "release";
})(envVersion || (envVersion = {}));
/**
 * @name AppCode
 */
var EAlipayPageAppCode;
(function (EAlipayPageAppCode) {
    /**
     * @description 打开支付宝扫一扫
     */
    EAlipayPageAppCode["alipayScan"] = "alipayScan";
    /**
     * @description 打开收到的红包页面
     */
    EAlipayPageAppCode["redPacket"] = "redPacket";
    /**
     * @description 打开爱攒油页面
     */
    EAlipayPageAppCode["collectOil"] = "collectOil";
    /**
     * @description 打开小程序快报生活号
     */
    EAlipayPageAppCode["tinyAppSHH"] = "tinyAppSHH";
    /**
     * @description 打开蚂蚁森林，其中 appParams 支持以下参数
     * - autoShowProps：是否打开背包，可空。
     */
    EAlipayPageAppCode["antForest"] = "antForest";
    /**
     * @description 打开蚂蚁庄园
     */
    EAlipayPageAppCode["antFarm"] = "antFarm";
    /**
     * @description 打开某只股票的详情页，其中 appParams 支持以下参数
     * - stockType：股票类型，ES-个股，MRI-指数
     * - market：所在市场，SH-沪市，SZ-深，A\O\N\USI - 美股市场，HK-港股市场
     * - symbol：股票代码；name：股票名称，需要encode
     */
    EAlipayPageAppCode["stockDetail"] = "stockDetail";
    /**
     * @description 打开支付宝付款码 (个人小程序暂不支持跳转付款码页面)
     */
    EAlipayPageAppCode["payCode"] = "payCode";
})(EAlipayPageAppCode || (EAlipayPageAppCode = {}));
